tools/ci_connector_ops/ci_connector_ops/qa_engine/validations.py

```diff
@@ -20,6 +29,14 @@ def url_is_reachable(url: str) -> bool:
 def is_appropriate_for_cloud_use(definition_id: str) -> bool:
     return definition_id not in INAPPROPRIATE_FOR_CLOUD_USE_CONNECTORS
 
+def is_eligible_for_cloud(connector_qa_data: pd.Series) -> bool:
+  if connector_qa_data["is_on_cloud"]:
+    return False
```

Comment:
Is there a case when something that wasn't on cloud before is now eligible for cloud because one of the previously passing checks now passes? Maybe I don't have the correct understanding of `is_on_cloud`

Reply:
No this flag is not dynamic.

We compute it by joining OSS and Cloud catalog: if the connector definition + version is not in the cloud catalog `is_on_cloud = false`

Reply:
How does a connector become `is_on_cloud` (and therefore `is_eligible_for_cloud`) for the first time? By being manually added to the cloud catalog?

Reply:
if `is_eligible_for_cloud is True` then this connector will be returned by the `get_connectors_eligible_for_cloud` function.
A future logic will create add this connector to cloud by:
- adding its definition id to the definitions mask on airbyte cloud
- open a PR with this change.
I started this work in https://github.com/airbytehq/airbyte/pull/21976

Reply:
As clarified - this is used solely to determine whether an OSS-only connector is eligible to be added to cloud. It doesn't apply to connectors that are already in cloud and determining whether they're still eligible (we don't currently implement any removal from cloud based on broken builds or anything like that). I was missing that `is_on_cloud` has to be `False`, and was thinking the opposite! 

LGTM 👍🏻